tests/DCPS/Compiler/xcdr/optional.idl

```diff
@@ -0,0 +1,24 @@
+module Optional {
+
+  typedef sequence<int32> seq_int_t;
+
+  struct TestStruct {
+    octet octet_field;
+  };
+
+  @appendable
+  struct OptionalMembers {
+    @optional boolean bool_field;
+    @optional short short_field;
+    @optional int32 int32_field;
+    @optional int64 int64_field;
+    @optional string str_field;
+    @optional seq_int_t seq_field;
+
+    @optional TestStruct struct_field;
+
+    // TODO(tyler) should this be possible? Maybe leave this for a future PR, typedefed sequences work
```

Comment:
`should this be possible?` -> yes, I think so.
Create a new GitHub Issue if this will be in a future PR.

Reply:
I'm working on this.